### ISSUE TITLE
Label is rendering different on Android and Desktop

### ISSUE THREAD

**Issue by ProZhar** [libgdx/libgdx#5246]
1. Label's width is defferent (better to see first and last lablel with text "1"):
on Desktop
![desktop](https://user-images.githubusercontent.com/9317340/40883380-049adac2-6704-11e8-9154-4fdb34b4f1a5.png)
on Android
![android](https://user-images.githubusercontent.com/9317340/40883381-0826b3aa-6704-11e8-88bf-fbe5118cdaa8.png)
2. Label's prefSize returns different value
on Desktop (getPrefWidht() = 41)
![1desktop](https://user-images.githubusercontent.com/9317340/40883401-9d6a873e-6704-11e8-9a1d-03e2191618c7.png)
on Android (getPrefWidht() = 27)
![1android](https://user-images.githubusercontent.com/9317340/40883412-e14fb492-6704-11e8-87e7-cf503748dfcd.png)
3. Label.cache.data has different varibles (seems android uses old spaceXadvance). Android marks missingGlyph, Desktop does not mark missingGlypsh
on Desktop
![2desktop](https://user-images.githubusercontent.com/9317340/40883425-40c602a0-6705-11e8-9183-c8b6718a8c97.png)
on Android
![2android](https://user-images.githubusercontent.com/9317340/40883427-47281aac-6705-11e8-8c81-f01215da91c6.png)
I try normal fonts and monospace fonts.
Font.setFixedWidthGlyphs() does not help

#### Issue details
_Please provide the details of your issue_

#### Reproduction steps/code
```java
Table rootTable = new Table():
rootTable.setFillParent(true);
stage.addActor(rootTable);
rootTable.validate();

Table oneLabelTable = new Table();
oneLabelTable.debugAll();
oneLabelTable.add(new Label("1$5$10 19$2$1", skin, "dollars"));
oneLabelTable.add().growX().height(168f);

Table severalLabelsTable = new Table();
severalLabelsTable.debugAll();
severalLabelsTable.add(new Label("1", skin, "dollars"));
severalLabelsTable.add(new Label("$5", skin, "dollars"));
severalLabelsTable.add(new Label("$10", skin, "dollars"));
severalLabelsTable.add(new Label(" ", skin, "dollars"));
severalLabelsTable.add(new Label("19", skin, "dollars"));
severalLabelsTable.add(new Label("$2", skin, "dollars"));
severalLabelsTable.add(new Label("$", skin, "dollars"));
severalLabelsTable.add(new Label("1", skin, "dollars"));
severalLabelsTable.add().growX().height(168f);

rootTable.add(oneLabelTable).growX().row();
rootTable.add(severalLabelsTable).growX().row();
```

### P.S. 
**I traied "Invalidate Caches/Restart..." on AS - did not help**
**Also I tried change libgdxVersion to "1.9.8" - in this case all good, but i need some features from snapshot**

#### Version of LibGDX and/or relevant dependencies
gdxVersion = '1.9.9-SNAPSHOT'
gradle files - default

#### Please select the affected platforms
- [x] Android
- [x] Windows 10 x64
- [ ] iOS (robovm)
- [ ] iOS (MOE)
- [ ] HTML/GWT
- [ ] Linux
- [ ] MacOS

**Comment by Darkyenus** [comment]
Looks like you have two libGDX versions mixed up, `spaceWidth` was [recently](beeef537dd7f76c72e124a5be674cd0c3b2b76f6) replaced with `spaceXAdvance`. These changes also fixed many GlyphLayout bugs, which explains different behavior.

Try to clean all caches and rebuild. Also check if the configured versions are the same for both backends.

**Comment by ProZhar** [comment]
@Darkyenus 
my main.gradle is default, android.gradle to
I traied "Invalidate Caches/Restart..." on AS - did not help

main.gradle
```gradle
buildscript {
    

    repositories {
        mavenLocal()
        mavenCentral()
        maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }
        jcenter()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:2.3.3'
    }
}

allprojects {
    apply plugin: "eclipse"
    apply plugin: "idea"

    version = '0.0.1'
    ext {
        appName = "Math"
        gdxVersion = '1.9.9-SNAPSHOT'
    }

    repositories {
        mavenLocal()
        mavenCentral()
        maven { url "https://oss.sonatype.org/content/repositories/snapshots/" }
        maven { url "https://oss.sonatype.org/content/repositories/releases/" }
    }
}

project(":desktop") {
    apply plugin: "java"


    dependencies {
        compile project(":core")
        compile "com.badlogicgames.gdx:gdx-backend-lwjgl:$gdxVersion"
        compile "com.badlogicgames.gdx:gdx-platform:$gdxVersion:natives-desktop"
        
    }
}

project(":android") {
    apply plugin: "android"

    configurations { natives }

    dependencies {
        compile project(":core")
        compile "com.badlogicgames.gdx:gdx-backend-android:$gdxVersion"
        natives "com.badlogicgames.gdx:gdx-platform:$gdxVersion:natives-armeabi"
        natives "com.badlogicgames.gdx:gdx-platform:$gdxVersion:natives-armeabi-v7a"
        natives "com.badlogicgames.gdx:gdx-platform:$gdxVersion:natives-arm64-v8a"
        natives "com.badlogicgames.gdx:gdx-platform:$gdxVersion:natives-x86"
        natives "com.badlogicgames.gdx:gdx-platform:$gdxVersion:natives-x86_64"
        
    }
}

project(":core") {
    apply plugin: "java"


    dependencies {
        compile "com.badlogicgames.gdx:gdx:$gdxVersion"
        
    }
}

tasks.eclipse.doLast {
    delete ".project"
}
```
android.gradle
```gradle
android {
    buildToolsVersion "27.0.3"
    compileSdkVersion 27
    sourceSets {
        main {
            manifest.srcFile 'AndroidManifest.xml'
            java.srcDirs = ['src']
            aidl.srcDirs = ['src']
            renderscript.srcDirs = ['src']
            res.srcDirs = ['res']
            assets.srcDirs = ['assets']
            jniLibs.srcDirs = ['libs']
        }

        instrumentTest.setRoot('tests')
    }
    packagingOptions {
        exclude 'META-INF/robovm/ios/robovm.xml'
    }
    defaultConfig {
        applicationId "com.com.com"
        minSdkVersion 19
        targetSdkVersion 27
        versionCode 1
        versionName "0.0.1"
    }
    buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }
    }
}


// called every time gradle gets executed, takes the native dependencies of
// the natives configuration, and extracts them to the proper libs/ folders
// so they get packed with the APK.
task copyAndroidNatives() { 
    file("libs/armeabi/").mkdirs();
    file("libs/armeabi-v7a/").mkdirs();
    file("libs/arm64-v8a/").mkdirs();
    file("libs/x86_64/").mkdirs();
    file("libs/x86/").mkdirs();

    configurations.natives.files.each { jar ->
        def outputDir = null
        if(jar.name.endsWith("natives-arm64-v8a.jar")) outputDir = file("libs/arm64-v8a")
        if(jar.name.endsWith("natives-armeabi-v7a.jar")) outputDir = file("libs/armeabi-v7a")        
        if(jar.name.endsWith("natives-armeabi.jar")) outputDir = file("libs/armeabi")
        if(jar.name.endsWith("natives-x86_64.jar")) outputDir = file("libs/x86_64")
        if(jar.name.endsWith("natives-x86.jar")) outputDir = file("libs/x86")
        if(outputDir != null) {
            copy {
                from zipTree(jar)
                into outputDir
                include "*.so"
            }
        }
    }
}

task run(type: Exec) {
    def path
    def localProperties = project.file("../local.properties")
    if (localProperties.exists()) {
        Properties properties = new Properties()
        localProperties.withInputStream { instr ->
            properties.load(instr)
        }
        def sdkDir = properties.getProperty('sdk.dir')
        if (sdkDir) {
            path = sdkDir
        } else {
            path = "$System.env.ANDROID_HOME"
        }
    } else {
        path = "$System.env.ANDROID_HOME"
    }

    def adb = path + "/platform-tools/adb"
    commandLine "$adb", 'shell', 'am', 'start', '-n', 'com.com.com/com.com.com.AndroidLauncher'
}

// sets up the Android Eclipse project, using the old Ant based build.
eclipse {
    // need to specify Java source sets explicitly, SpringSource Gradle Eclipse plugin
    // ignores any nodes added in classpath.file.withXml
    sourceSets {
        main {
            java.srcDirs "src", 'gen'
        }
    }

    jdt {
        sourceCompatibility = 1.6
        targetCompatibility = 1.6
    }

    classpath {
        plusConfigurations += [ project.configurations.compile ]        
        containers 'com.android.ide.eclipse.adt.ANDROID_FRAMEWORK', 'com.android.ide.eclipse.adt.LIBRARIES'       
    }

    project {
        name = appName + "-android"
        natures 'com.android.ide.eclipse.adt.AndroidNature'
        buildCommands.clear();
        buildCommand "com.android.ide.eclipse.adt.ResourceManagerBuilder"
        buildCommand "com.android.ide.eclipse.adt.PreCompilerBuilder"
        buildCommand "org.eclipse.jdt.core.javabuilder"
        buildCommand "com.android.ide.eclipse.adt.ApkBuilder"
    }
}

// sets up the Android Idea project, using the old Ant based build.
idea {
    module {
        sourceDirs += file("src");
        scopes = [ COMPILE: [plus:[project.configurations.compile]]]        

        iml {
            withXml {
                def node = it.asNode()
                def builder = NodeBuilder.newInstance();
                builder.current = node;
                builder.component(name: "FacetManager") {
                    facet(type: "android", name: "Android") {
                        configuration {
                            option(name: "UPDATE_PROPERTY_FILES", value:"true")
                        }
                    }
                }
            }
        }
    }
}

```

**Comment by cypherdare** [comment]
@ProZhar, was the issue resolved after switching to 1.9.9 or 1.9.10?

**Comment by MrStahlfelge** [comment]
Closed as the original reporter did not react for more than a year and the problem can't be confirmed.
